test/new-e2e/pkg/utils/e2e/pulumi_provider.go

```diff
@@ -0,0 +1,60 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package e2e
+
+import (
+	"context"
+	"testing"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+)
+
+var _ InfraProvider[int] = (*PulumiProvider[int])(nil)
+
+// PulumiProvider leverages pulumi to create and manage testing infrastructure
+type PulumiProvider[Env any] struct {
+	envFactory func(ctx *pulumi.Context) (*Env, error)
+	configMap  runner.ConfigMap
+
+	stackManager *infra.StackManager
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
`StackManager` could move to a package specific to `pulumi` like `pulumiinfra` or `e2epulumi`